funsor/ops/builtin.py

```diff
@@ -144,10 +143,7 @@ class LogOp(TransformOp):
 
 @LogOp
 def log(x):
-    if isinstance(x, bool) or (isinstance(x, np.ndarray) and x.dtype == 'bool'):
```

Comment:
note I've refactored this conditional into two different patterns: one for default values, one for numpy values (`np.array` and `np.generic`).